docs/building-modules.md

```diff
@@ -132,13 +132,76 @@ individual methods in
 
 - [DOMException](https://developer.mozilla.org/en-US/docs/Web/API/DOMException)
 
+### libp2p
+
+Zinnia comes with a built-in libp2p node based on
+[rust-libp2p](https://github.com/libp2p/rust-libp2p). The node is shared by all
+Station Modules running on Zinnia. This way we can keep the number of open
+connections lower and avoid duplicate entries in routing tables.
+
+The initial version comes with a limited subset of features. We will be adding
+more features based on feedback from our users (Station Module builders).
+
+#### Networking stack
+
+- Transport: `tcp` using system DNS resolver
+- Multistream-select V1
+- Authentication: `noise` with `XX` handshake pattern using X25519 DH keys
+- Stream multiplexing: both `yamux` and `mplex`
+
+#### `Zinnia.peerId`
+
+Type: `string`
+
+Return the peer id of Zinnia's built-in libp2p peer. The peer id is ephemeral,
+Zinnia generates a new peer id every time it starts.
+
+#### `Zinnia.requestProtocol(remoteAddress, protocolName, requestPayload)`
+
+```ts
+requestProtocol(
+  remoteAddress: string,
+  protocolName: string,
+  requestPayload: Uint8Array,
+): Promise<PeerResponse>;
+```
+
+Dial a remote peer identified by the `remoteAddress` and open a new substream
+for the protocol identified by `protocolName`. Send `requestPayload` and read
+the response payload.
+
+The function returns a promise that resolves with a readable-stream-like object.
+Exact shape will be determined during implementation of this API. We want to be
```

Comment:
What do you mean by exact shape will be determined? Does it mean that extra keys will be added in the future? At first I read it as if this API wasn't implemented at all 🤔

Reply:
You are right! I forgot to update this text yesterday; I did it in the morning:

https://github.com/filecoin-station/zinnia/pull/48/commits/e125eae74819a90427beadd313fb4791312c6d74